packages/core/src/components/forms/controls.tsx

```diff
@@ -39,10 +40,15 @@ export class Control<P extends IControlProps> extends React.Component<React.HTML
     // generates control markup for given input type.
     // optional inputRef in case the component needs reference for itself (don't forget to invoke the prop!).
     protected renderControl(type: "checkbox" | "radio", typeClassName: string, inputRef = this.props.inputRef) {
+        const className = classNames(
+            Classes.CONTROL,
+            typeClassName,
+            this.props.className,
```

Comment:
`this.props.className` should always come last in `classNames()` so user classes take precedence.